draft-ietf-ppm-dap.md

```diff
@@ -1298,17 +1320,21 @@ The Leader and Helper initialization behavior is detailed below.
 
 #### Leader Initialization {#leader-init}
 
-The Leader begins the aggregate initialization phase with the set of candidate
-reports as follows:
+The Leader begins the aggregate initialization by sampling a fresh
+AggregationJobID.
+
+Next, for each report in the candidate set, it checks if the report ID
+corresponds to a report ID it has previously stored for this task. If so, it
+marks the report as invalid and removes it from the candidate set.
 
-1. Generate a fresh AggregationJobID.
-1. Decrypt the input share for each report share as described in
-   {{input-share-decryption}}.
-1. Check that the resulting input share is valid as described in
-   {{input-share-validation}}.
+At this point, the Leader MAY store the ID of each report in the candidate set.
+It MUST do so at the end of the continuation phase
+({{aggregation-leader-continuation}}).
 
-If any step invalidates the report, the Leader rejects the report and removes
-it from the set of candidate reports.
+Next, the Leader decrypts each of its report shares as described in
+{{input-share-decryption}}, then checks input share validity as described in
+{{input-share-validation}}. If either step invalidates the report, the Leader
+rejects the report and removes it from the set of candidate reports.
```

Comment:
I think we should have some text suggesting that the `Finished(out_share)` case might require replay-checking, around L1473 of this snaphost, similarly to what is written in the Leader Continuation section for the `Finished` state handling.

Reply:
Done.